testsuite/compat/src/test/java/org/jboss/as/test/compat/jpa/hibernate/transformer/AbstractVerifyHibernate51CompatibilityTestCase.java

```diff
@@ -462,4 +466,43 @@ public void testAbstractSingleColumnStandardBasicTypeReplaceExtended() {
         }
 
     }
+
+    public static class EnableHibernateBytecodeTransformerSetupTask implements ServerSetupTask {
+        private static final ModelNode PROP_ADDR_ENABLETRANSFORMER = new ModelNode()
+                .add("system-property", "Hibernate51CompatibilityTransformer");
+
+        @Override
+        public void setup(ManagementClient managementClient, String s) throws Exception {
+            ModelNode op = Operations.createAddOperation(PROP_ADDR_ENABLETRANSFORMER);
+            op.get("value").set("true");
+            managementClient.getControllerClient().execute(op);
+        }
+
+        @Override
+        public void tearDown(ManagementClient managementClient, String s) throws Exception {
+            ModelNode op = Operations.createRemoveOperation(PROP_ADDR_ENABLETRANSFORMER);
+            managementClient.getControllerClient().execute(op);
+            ServerReload.executeReloadAndWaitForCompletion(managementClient.getControllerClient());
```

Comment:
If the system property was static a reload wouldn't be required. Maybe there's no real way around that though.

Reply:
Maybe, I'm not sure.

Reply:
It could be a performance issue, but there's not feedback when removing a property that a reload would be required. It could just be documented though. Just more of an idea than an actual issue with the PR :)